### ISSUE TITLE
Artificial Intelligence and Big Data: Assignment01, 2024-05-31

### ISSUE THREAD

**Issue by fkdl0048** [fkdl0048/Liberal_Arts_Subject#108]
import numpy as np
import pandas as pd
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from google.colab import files

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 1단계: 데이터 불러오기
uploaded = files.upload()

train_data = pd.read_csv('train_data.csv')
test_data = pd.read_csv('test_data.csv')

# 데이터 확인
print("Train Data Head:")
print(train_data.head())

print("Test Data Head:")
print(test_data.head())

# 2단계: 데이터 준비
X_train = train_data.iloc[:, :-1]
y_train = train_data.iloc[:, -1]

X_test = test_data.iloc[:, :-1]
y_test = test_data.iloc[:, -1]

# 정보 출력
print("X_train Information:")
print(X_train.info())

print("X_test Information:")
print(X_test.info())

# 3단계: 모델 구성 및 학습
model = Sequential()
model.add(Dense(64, input_dim=11, activation='relu'))
model.add(Dense(32, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

history = model.fit(X_train, y_train, epochs=100, batch_size=10, validation_data=(X_test, y_test), verbose=1)

# Epoch에 따른 결과 출력
for i in range(len(history.history['accuracy'])):
    print(f"Epoch {i+1}: Loss = {history.history['loss'][i]}, Accuracy = {history.history['accuracy'][i]}, Val Loss = {history.history['val_loss'][i]}, Val Accuracy = {history.history['val_accuracy'][i]}")

# 4단계: 성능 평가
loss, accuracy = model.evaluate(X_test, y_test)
print(f'Test Accuracy: {accuracy:.4f}')
